src/Raven.Client/Documents/Operations/AI/AiAgent/AddOrModifyAiAgentOperation.cs

```diff
@@ -0,0 +1,91 @@
+﻿using System;
+using System.Diagnostics.CodeAnalysis;
+using System.Net.Http;
+using Raven.Client.Documents.Conventions;
+using Raven.Client.Http;
+using Raven.Client.Json;
+using Raven.Client.Json.Serialization;
+using Raven.Client.Util;
+using Sparrow.Json;
+
+namespace Raven.Client.Documents.Operations.AI.AiAgent
+{
+    internal class AddOrModifyAiAgentOperation : AddOrModifyAiAgentOperation<object>
+    {
+        public AddOrModifyAiAgentOperation(string name, AiAgentConfiguration configuration) : base(name, configuration)
+        {
+            if (string.IsNullOrEmpty(configuration.OutputSchema))
+                throw new ArgumentException("OutputSchema cannot be null or empty.", nameof(configuration.OutputSchema));
+        }
+    }
+
+    internal class AddOrModifyAiAgentOperation<T> : IMaintenanceOperation<AiAgentConfigurationResult> where T : new()
+    {
+        private readonly string _name;
+        private readonly AiAgentConfiguration _configuration;
+
+        public AddOrModifyAiAgentOperation(string name, AiAgentConfiguration configuration)
+        {
+            _name = name ?? throw new ArgumentNullException(nameof(name));
+            _configuration = configuration ?? throw new ArgumentNullException(name);
+            
+        }
+
+        protected void Initialize()
+        {
+            if (_configuration.OutputSchema == null)
+            {
+                using (var context = JsonOperationContext.ShortTermSingleUse())
+                {
+                    _configuration.OutputSchema = DocumentConventions.Default.Serialization.DefaultConverter.ToBlittable(new T(), context).ToString();
```

Comment:
That is meant to specify what? What is it that you are doing here?

Reply:
allow to have a typed schema e.g.
```
        public class OutputSchema
        {
            public string Answer = "Answer to the user question";

            public bool Relevant = true;

            public List<string> RelevantOrdersId = ["The order ids relevant to the query or response"];

            public List<string> MatchingProductsId = ["All the product ids referenced either by the user or the system"];
        }
```